bigbluebutton-html5/imports/ui/components/dropdown/component.jsx

```diff
@@ -67,21 +67,39 @@ class Dropdown extends Component {
     this.handleHide = this.handleHide.bind(this);
     this.handleToggle = this.handleToggle.bind(this);
     this.handleWindowClick = this.handleWindowClick.bind(this);
+    this.componentDidUpdate = this.componentDidUpdate.bind(this);
   }
 
   componentWillUpdate(nextProps, nextState) {
     return nextState.isOpen ? screenreaderTrap.trap(this.dropdown) : screenreaderTrap.untrap();
   }
 
+  componentDidUpdate(prevProps, prevState, event) {
+    const {
+      userDropdownOpen,
+      closeUserDropdown,
+      showEmojiMenu,
```

Comment:
I do not think it's a good idea the dropdown to know this, why not create a component that controls the toggle function? instead of the dropdown having to know when it has emoji menu in the dropdown and the events to trigger i